### PR TITLE
test: create basic jest config

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Creates a basic "boilerplate" Jest config with `jest init` command

## Context:

When you run the Jests, you now get a nice overview of your test coverage on the command line.

I've tried to set things up sensibly for what we have now. We can throw this away and start fresh if needed later on anyways.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // An array of glob patterns indicating a set of files for which coverage information should be collected
+  // collectCoverageFrom: undefined,
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: "coverage",
```
Comment (viceice) [inline jest.config.js:26]: Thisis the default value.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+  // An array of regexp pattern strings used to skip coverage collection
+  // coveragePathIgnorePatterns: [
+  //   "\\\\node_modules\\\\"
+  // ],
+
+  // Indicates which provider should be used to instrument code for coverage
+  coverageProvider: "v8",
```
Comment (viceice) [inline jest.config.js:34]: This is not good for typescript yet. i think this will be more usable when next node lts becomes released

**Review by viceice — COMMENTED** [review]
Can we remove all comments and default values? I don't see any value for it. The jest documentation is very good about available settings.

**Comment by HonkingGoose** [comment]
@viceice I've removed the comments, except for the top one that explains where to find the docs.

I don't understand what changes you want me to make, so can you maybe comment/suggest some more? 😉

**Comment by viceice** [comment]
updated comments 🙃

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
+ */
+
+module.exports = {
+  clearMocks: true,
+  collectCoverage: true,
```
Comment (viceice) [inline jest.config.js:8]:
```suggestion
  cacheDirectory: '.cache/jest',
  collectCoverage: true,
  coverageDirectory: './coverage',
  coverageReporters: ['html', 'json', 'text-summary'],
  testEnvironment: 'node',
  testRunner: 'jest-circus/runner',
```

Added back some settings we also override in renovate repo.

**Comment by HonkingGoose** [comment]
I was wondering if we needed to install the `jest-circus/runner` package as a devDependency, but it seems we do not need to do that (anymore) as it is the default test runner for Jest 27.

Quote from [`jest-circus` package readme](https://github.com/facebook/jest/tree/master/packages/jest-circus#installation):

> ## Installation
> 
> > Note: As of Jest 27, `jest-circus` is the default test runner, so you do not have to install it to use it.

**Comment by viceice** [comment]
> I was wondering if we needed to install the `jest-circus/runner` package as a devDependency, but it seems we do not need to do that (anymore) as it is the default test runner for Jest 27.
> 
> Quote from [`jest-circus` package readme](https://github.com/facebook/jest/tree/master/packages/jest-circus#installation):
> 
> > ## Installation
> > 
> > > Note: As of Jest 27, `jest-circus` is the default test runner, so you do not have to install it to use it.

You are right, i think they also switched to node env by default?

**Review by viceice — APPROVED** [review]

**Comment by HonkingGoose** [comment]
> You are right, i think they also switched to node env by default?

According to the Jest 27 blog post, they switched the default to node env:

> For this reason, we are changing the default test environment from "jsdom" to "node".

https://jestjs.io/blog/2021/05/25/jest-27#flipping-defaults

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ * https://jestjs.io/docs/configuration
+ */
+
+module.exports = {
+  clearMocks: true,
+  cacheDirectory: ".cache/jest",
+  collectCoverage: true,
+  coverageDirectory: "./coverage",
+  coverageReporters: ["html", "json", "text-summary"],
+  testEnvironment: "node",
```
Comment (HonkingGoose) [inline jest.config.js:12]:
```suggestion

```

Do we need to be explicit with our testEnvironment, or can we rely on the Jest 27 default to `node`?

Comment (viceice) [reply jest.config.js:12]: Sure, we can rely on default.

File: jest.config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  testRunner: "jest-circus/runner",
```
Comment (viceice) [inline jest.config.js:13]:
```suggestion
```
Is default too.

**Pull request merged**